### PR TITLE
Add simple test module to change kernel

### PR DESCRIPTION
**Pull request by richiejp**
Unlike install_kotd and update_kernel which do some weird complicated stuff, this just
removes the currently installed kernels and installs a new one specified with
CHANGE_KERNEL_REPO and CHANGE_KERNEL_PKG.

### CODE REVIEW

**Comment by richiejp** [comment]
Effected by https://github.com/os-autoinst/os-autoinst/pull/1007

**Comment by richiejp** [comment]
http://10.160.67.78/tests/669

File: lib/utils.pm [hunk 1]
```diff
@@ -1291,4 +1293,20 @@ sub reconnect_s390 {
     }
 }
 
+sub kernel_packages {
```
Comment (asmorodskyi) [inline lib/utils.pm]: there is a high chance that this function would be used exclusively by kernel tests. what if you will create kernel-utils and move it there?  current lib/utils is too big. I already did attempts to solve this by introducing version_utils, but there is still a lot of place for improvement. So let's at least not increase this big file with something which is used only in one file or something which used in some certain area ( e.g. kernel, nfv  etc. )

File: tests/kernel/change_kernel.pm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    my @packages = kernel_packages;
+    zypper_call('-n rm ' . join(' ', @packages), exitcode => [0, 104], dumb_term => 1);
+
+    # Install the new kernel
+    zypper_ar($repo, 'change-kernel') if ($repo);
+    zypper_call("in --force-resolution --force --replacefiles --repo change-kernel $pkg",
+        dumb_term => 1);
+
+    # Reboot into the new kernel
+    select_console('root-console');
```
Comment (asmorodskyi) [inline tests/kernel/change_kernel.pm]: reboot not working  in case of virtio_console selected ?

File: tests/kernel/change_kernel.pm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    my $self = shift;
+    my $repo = get_var('CHANGE_KERNEL_REPO');
+    my $pkg  = get_var('CHANGE_KERNEL_PKG') || 'kernel-default';
+
+    $self->wait_boot;
+    select_virtio_console();
+
+    # Avoid conflicts by removing any existing kernels
+    my @packages = kernel_packages;
+    zypper_call('-n rm ' . join(' ', @packages), exitcode => [0, 104], dumb_term => 1);
```
Comment (asmorodskyi) [inline tests/kernel/change_kernel.pm]: currently you using kernel_packages function in three places. In all 3 places you need this function to get list of kernel packages to delete. For it looks more reasonable to replace this two lines with function `cleanup_kernel` which will contains code of kernel_packages and also line below which actually delete them

**Comment by richiejp** [comment]
from RPM: http://10.160.67.78/tests/682
from repo: http://10.160.67.78/tests/683

**Pull request merged**